File: testdata/jennies/builders/struct_with_defaults/PHPBuilder/src/StructWithDefaults/StructBuilder.php
<?php

namespace Grafana\Foundation\StructWithDefaults;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\StructWithDefaults\Struct>
 */
class StructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\StructWithDefaults\Struct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\StructWithDefaults\Struct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\StructWithDefaults\Struct
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\StructWithDefaults\NestedStruct> $allFields
     */
    public function allFields(\Grafana\Foundation\Cog\Builder $allFields): static
    {
        $allFieldsResource = $allFields->build();
        $this->internal->allFields = $allFieldsResource;
    
        return $this;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\StructWithDefaults\NestedStruct> $partialFields
     */
    public function partialFields(\Grafana\Foundation\Cog\Builder $partialFields): static
    {
        $partialFieldsResource = $partialFields->build();
        $this->internal->partialFields = $partialFieldsResource;
    
        return $this;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\StructWithDefaults\NestedStruct> $emptyFields
     */
    public function emptyFields(\Grafana\Foundation\Cog\Builder $emptyFields): static
    {
        $emptyFieldsResource = $emptyFields->build();
        $this->internal->emptyFields = $emptyFieldsResource;
    
        return $this;
    }

    public function complexField(unknown $complexField): static
    {
        $this->internal->complexField = $complexField;
    
        return $this;
    }

    public function partialComplexField(unknown $partialComplexField): static
    {
        $this->internal->partialComplexField = $partialComplexField;
    
        return $this;
    }

}

File: testdata/jennies/builders/struct_with_defaults/JavaBuilder/struct_with_defaults/StructBuilder.java
package struct_with_defaults;


public class StructBuilder implements cog.Builder<Struct> {
    protected final Struct internal;
    
    public StructBuilder() {
        this.internal = new Struct();
    }
    public StructBuilder allFields(cog.Builder<NestedStruct> allFields) {
    NestedStruct allFieldsResource = allFields.build();
        this.internal.allFields = allFieldsResource;
        return this;
    }
    
    public StructBuilder partialFields(cog.Builder<NestedStruct> partialFields) {
    NestedStruct partialFieldsResource = partialFields.build();
        this.internal.partialFields = partialFieldsResource;
        return this;
    }
    
    public StructBuilder emptyFields(cog.Builder<NestedStruct> emptyFields) {
    NestedStruct emptyFieldsResource = emptyFields.build();
        this.internal.emptyFields = emptyFieldsResource;
        return this;
    }
    
    public StructBuilder complexField(Object complexField) {
        this.internal.complexField = complexField;
        return this;
    }
    
    public StructBuilder partialComplexField(Object partialComplexField) {
        this.internal.partialComplexField = partialComplexField;
        return this;
    }
    public Struct build() {
        return this.internal;
    }
}
